test/integration/cli/cli-test.coffee

```diff
@@ -185,64 +197,57 @@ describe 'CLI', ->
         it 'should not execute any transaction', ->
           assert.deepEqual runtimeInfo.server.requestCounts, {}
 
-
       describe "and handler is killed during execution", ->
-        dreddCommandInfo = undefined
-        serverRuntimeInfo = undefined
+        runtimeInfo = undefined
 
         before (done) ->
           app = createServer()
-
           app.get '/machines', (req, res) ->
             killAll('endless-nosigterm.+[^=]foo/bar/hooks', (err) ->
               done err if err
               res.json([{type: 'bulldozer', name: 'willy'}])
             )
 
           # TCP server echoing transactions back
-          hookServer = net.createServer (socket) ->
-            socket.on 'data', (data) ->
-              socket.write data
-
-          hookServer.listen DEFAULT_HOOK_HANDLER_PORT, ->
-            server = app.listen DEFAULT_SERVER_PORT, (err, info) ->
-              serverRuntimeInfo = info
-              runDreddCommand [
-                './test/fixtures/single-get.apib'
-                "http://127.0.0.1:#{DEFAULT_SERVER_PORT}"
-                '--server=./test/fixtures/scripts/endless-nosigterm.sh'
-                '--language=./test/fixtures/scripts/endless-nosigterm.sh'
-                '--hookfiles=foo/bar/hooks'
-                '--server-wait=0'
-              ], (err, info) ->
-                dreddCommandInfo = info
-                server.close()
-
-            server.on 'close', ->
-              hookServer.close()
-              done()
+          hookHandler = net.createServer (socket) ->
+            socket.on 'data', (data) -> socket.write data
+            socket.on 'error', (err) -> console.error err
+
+          args = [
+            './test/fixtures/single-get.apib'
+            "http://127.0.0.1:#{DEFAULT_SERVER_PORT}"
+            "--server=#{COFFEE_BIN} ./test/fixtures/scripts/endless-nosigterm.coffee"
```

Comment:
In all the tests in this section the `endless-nosigterm` process is used to imitate a running server process. It's basically a placeholder to pass as an argument and for Dredd to spawn and then terminate. But in fact, Dredd requests `http://127.0.0.1:#{DEFAULT_SERVER_PORT}`, which is where the `app = createServer()` above serves. So Dredd is feeded with a dummy process to work with, while any of its requests target an `express` server created in the test.

In this particular test, the same goes for the hook handler process. We feed Dredd with a dummy `endless-nosigterm` process, but in fact the actual socket connection goes to the receiver created on the `hookHandler = net.createServer (socket) ->` line.

The test checks a situation when Dredd is successfully launched with valid server and hook handler processes, but during server request the hook handler is, for some reason, forcefully killed.